eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Address/CompactAddr.hs

```diff
@@ -60,8 +60,7 @@ propDecompactShelleyLazyAddr = do
       keyHash1 = unsafeGetHash . CA.decompactAddr . mangle . CA.compactAddr $ addr
    in pure $ keyHash0 == keyHash1
 
--- | TODO This property test is failing to find a discrepancy that was found on mainnet.
-propIsBootstrapRedeemer :: Addr crypto -> Property
+propIsBootstrapRedeemer :: CC.Crypto crypto => Addr crypto -> Property
```

Comment:
does this constraint fix the test not covering a case that was on mainnet? is this some overlapping instance or something

Reply:
The `Crypto` constraint is necessary due to the fact that the `isBootstrapRedeemer` that accepts compact addresses now sometimes needs to decompact an address, which requires that constraint. This test checks two separate functions.

From what I've seen so far in the codebase we don't have any overlapping instances, otherwise I'd immediately advocate for their removal, it is a bad practice most of the time.